### ISSUE TITLE
Add inverse/not habbits

### ISSUE THREAD

**Issue by c33s** [iSoron/uhabits#351]
if a person want to stop something, like eating chocolate, playing computer, watching tv, smoking, drinking,... it would be awesome if it is possible to only mark the days where the habbit of **not doing** it was broken.

like if i want to stop eating sweets and i am quite good at not thinking about them and the tracking of not doing it, reminds me of wanting them.
for such things an inverse tracking would be awesome. only track the "bad" days

| mon | tue | wed | thu | fri | sat | sun |                                                                  |
|-----|-----|-----|-----|-----|-----|-----|------------------------------------------------------------------|
| x   | x   | x   |     | x   | x   | x   | marks on the day where i not having done what i don't want to do |
| vs  |     |     |     |     |     |     |                                                                  |
|     |     |     | x   |     |     |     | mark on the day when having done what i dont want                |

**Comment by paulrm** [comment]
I love it! <3

**Comment by tickTackHack** [comment]
To implement this may simply only take an option to inverse the grammar/phrasing rather than a technical change. So for example, a switch to change wording such as "Habit strength" to "Abstinence strength" or similar.

**Comment by iSoron** [comment]
Thank you for the suggestion, but I have decided a while ago not to implement this feature. Please, see #15 for the rationale.

**Comment by c33s** [comment]
@iSoron 

>Please, feel free to try to change my view.

please rethink your point of view. i am not allone with this feature wish.
rephrasing allone is not the key it is also relevant how much interaction with the app is needed. sometimes it is the key of not thinking about something, so not beeing forced to open an app and add a check every day.

in #15 you suggested the positive rephrasing:

> "Did you smoke today?"

vs

>"Did you have a smoke-free day today?".

this need me to micromanage the habbit each day. this can push a person in two different directions. 

the first direction is that one you want to push, having a positive feeling of setting a check for the day where the person has done it, reached the goal. so the check is the reward. so it is a good motivation.

on the other side we have the other direction. having to put a check on each day, forces a person to be confronted with the habbit they want to loose, every day. this can also result in bringing the habbit the person wants to loose back in their head and thoughts. this is not positive.

persons are different, for me personal it makes no real sense to force a user to use this awesome app in a specific way. for `x%` people your approach of having a check each day is the right one, for the other `y%`  the other way is the right one for them.
as this is software and not a building, it is not necessary to decide `platform roof` OR `pitched roof`, `positive habbit` OR `negative habbit` it is possible to support both and let the individual user decide.

if you see it from the basic user experience/usability. it is much better to only add one check of the few days where, as in your example, smoking happend. if you rephrase it to "smoke free", i have to micromanage this habbit every day. it makes me think every day about this habbit, it remember me, that i miss something,... 

another point for the negative habbit would be that lazy people can be motivated if they don't have to open the app to add a check every day if they follow the habbit of not doing it. if they follow the habbit, they have to go into the app and set the check.

**Comment by AeliusSaionji** [comment]
Just use the phrasing, "did you smoke today?" and treat the data collected as a measure of your failure. Use the color red for such "negative habits".

This is literally exactly what you described in the first post- there is no need to change any code at all to achieve this...

**Comment by ke352802081770314** [comment]
I was initially also in favour of this suggestion but after using the app for a while detected that it is not needed and the arguments that seem to support it can be easily dismissed.
Despite the fact that this issue is closed, I would like to comment to make it easier for supporters to see the alternatives.

For any kind of abstinence tracking I would strongly recommend to activate the option "display question mark for missing data".
 
My original objection was that to state "avoided sweets today" I would have to wait until the day is over to know whether to mark it as done, and then have to honestly and not forgetfully review the day whether there were any lapses, while with a negative habit I could record the lapse immediately when it occurred.
But with "?" enabled I can just press it twice when a lapse occurred to make the ? into an X when I lapsed (when I would have marked it as checked if a negative habit existed). After the day I can easily tick as success what I haven't marked as a fail.

This also addresses the objection of "too many interactions": 
Firstly, if the app is (also) used for habit building as designed, you will want to have at least daily interactions to mark your accomplishments, and checking a "refrained from vice X" (or positive rephrasing) will just increase your number of successes and feel good.
Secondly, if you don't interact with the app that often or if you placed all the abstinence stuff at the bottom of the list so it's out of sight and don't regularly scroll down to it (the "other direction" explained by @c33s above), you can later fill all the question mark streaks with ticks when you want to look at the score you achieved. I think this clears the "micro-manage" objection.

I would advise against what @AeliusSaionji suggests, i.e. to add habits with inverted meaning. I think it confuses the mind if you need to differentiate which lines should have a high score and which lines should have a low score, the whole concept of score becomes muddled by doing this. (Also, by aiming for a low score, you glorify the past: Loop Habit Tracker rightly considers unknown data as no success, because there is no evidence of achievement. If you switch to this app and have a past record, you can go back in the timeline and fill it in. If you model abstinence by the absence of ticking a habit, you make the past into perfect success, but if that were the case, you would hardly feel that it is necessary to track it now.)

Some arguments are that some statements are difficult to invert. Maybe some examples were trying too hard and changed the meaning. Clearly "didn't eat junk food today" isn't the same as "ate a home-cooked meal today" - if you decided on a day of fasting, neither applies, if you ate a burger and fries for lunch or during a night out and a home-cooked dinner on the same day, both applies. So the goal is "avoided junk food today". Avoiding a vice is a good thing just like performing a virtue. So "avoid" is not a negative word in general and may well be used in a goal. We need not (and I think should not) transform our abstinence goal into something different. Vague terms like "eat healthy" that can be re-interpreted as the situation makes convenient are not good because the goal loses its edge. Just like KPIs in business, it is better if our goals are measurable or clearly decideable.

**Comment by eshimoniak** [comment]
> For any kind of abstinence tracking I would strongly recommend to activate the option "display question mark for missing data".

I've been doing this for a while with my own negative habits, but its ended up having an unhealthy side effect.

If I slip up early in the day, I'll put an `X` on that day right then. If I am tempted again later in the day, the record shows no difference between 1 slip up and 100 slip ups so long as they occur during the same day. I end up in the mindset of "I've already failed today, and even if I do a good job the rest of today, on the record itself it will not make a difference," which leads a single mistake to balloon into many.

To use the junk food example, I would say that a eating single candy bar should be treated differently from binging several thousand calories worth of sugar in one sitting.

**Comment by goodwinnk** [comment]
> Just use the phrasing, "did you smoke today?" and treat the data collected as a measure of your failure. Use the color red for such "negative habits".

Do it for a while — red color, inverted meaning "a day without...". It still doesn't work very well. Mostly it is fine, and you see the days when you've slipped with a habit. But you also get a cheerful animation on check, best streaks of failures, and an inverted score chart which is undesirable.
